portfolio/src/main/webapp/script.js

```diff
@@ -16,8 +16,19 @@
  * Adds a random greeting to the page.
  */
 
-async function getHelloMessage(){
+async function getQuote(){
     const response = await fetch("/data");
-    const msg = await response.text();
-    document.getElementById("msg").innerText = msg;
+    const quote = await response.text();
+    document.getElementById("quote-container").innerText = quote;
+}
+async function getMessages(){
+    const list = document.getElementById("message-list");
+    const response = await fetch("/messages");
+    const messages = await response.json();
+    
+    messages.forEach(function(item){
+        const listElement = document.createElement("li");
+        listElement.innerHTML = item;
```

Comment:
If I left a message: "</li><script>alert("hi")</script>", what would happen to you when you load your site? (Try it out and see!)

Generally you want to be very careful when using innerHTML, as use of this API will tend to expose you to a class of security vulnerabilities called XSS.

See also: https://xss-game.appspot.com/